src/meta/client/MetaClient.cpp

```diff
@@ -995,5 +1003,181 @@ folly::Future<StatusOr<int64_t>> MetaClient::balance() {
+    item.set_name(name);
+    item.set_type(type);
+    item.set_mode(cpp2::ConfigMode::MUTABLE);
+    item.set_value(value);
+
+    cpp2::SetConfigReq req;
+    req.set_item(item);
+    return getResponse(std::move(req), [] (auto client, auto request) {
+                return client->future_setConfig(request);
+            }, [this] (cpp2::ExecResp&& resp) -> decltype(auto) {
+                return resp.code == cpp2::ErrorCode::SUCCEEDED;
+            }, true);
+}
+
+folly::Future<StatusOr<std::vector<cpp2::ConfigItem>>>
+MetaClient::listConfigs(const cpp2::ConfigModule& module) {
+    cpp2::ListConfigsReq req;
+    req.set_module(module);
+    return getResponse(std::move(req), [] (auto client, auto request) {
+                return client->future_listConfigs(request);
+            }, [this] (cpp2::ListConfigsResp&& resp) -> decltype(auto) {
+                return std::move(resp).get_items();
+            });
+}
+
+void MetaClient::setGflagsModule(const cpp2::ConfigModule& module) {
+    if (module != cpp2::ConfigModule::UNKNOWN) {
+        gflagsModule_ = module;
+        return;
+    }
+
+    // get current process according to gflags pid_file
+    gflags::CommandLineFlagInfo pid;
+    if (gflags::GetCommandLineFlagInfo("pid_file", &pid)) {
+        auto defaultPid = pid.default_value;
+        if (defaultPid.find("nebula-graphd") != std::string::npos) {
+            gflagsModule_ = cpp2::ConfigModule::GRAPH;
+        } else if (defaultPid.find("nebula-storaged") != std::string::npos) {
+            gflagsModule_ = cpp2::ConfigModule::STORAGE;
+        } else if (defaultPid.find("nebula-metad") != std::string::npos) {
+            gflagsModule_ = cpp2::ConfigModule::META;
+        } else {
+            LOG(ERROR) << "Should not reach here";
+        }
+    } else {
+        LOG(ERROR) << "Should not reach here";
+    }
+}
+
+void MetaClient::loadCfgThreadFunc() {
+    loadCfg();
+    addLoadCfgTask();
+}
+
+void MetaClient::loadCfg() {
+    // only load current module's config is enough
+    auto ret = listConfigs(gflagsModule_).get();
+    if (ret.ok()) {
+        // if we load config from meta server successfully, update gflags and set configReady_
+        auto tItems = ret.value();
+        std::vector<ConfigItem> items;
+        for (const auto& tItem : tItems) {
+            items.emplace_back(toConfigItem(tItem));
+        }
+        MetaConfigMap metaConfigMap;
+        for (auto& item : items) {
+            std::pair<cpp2::ConfigModule, std::string> key = {item.module_, item.name_};
+            metaConfigMap.emplace(std::move(key), std::move(item));
+        }
+        {
+            // For any configurations that is in meta, update in cache to replace previous value
+            folly::RWSpinLock::WriteHolder holder(configCacheLock_);
+            for (const auto& entry : metaConfigMap) {
+                auto& key = entry.first;
+                if (metaConfigMap_.find(key) == metaConfigMap_.end() ||
```

Comment:
the if will traversing 3 times the map if key is exist in metaConfigMap_.

Reply:
fixed